### ISSUE TITLE
Query: ordering by the same property twice fails if that property is accessed thru navigation

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFrameworkCore#10133]
Found during verification of #10045 

Repro:

```cs
using System;
using System.Collections.Generic;
using System.Linq;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Logging;

namespace EFSampleApp
{
    public class Program
    {
        public static void Main(string[] args)
        {
            using (var db = new MyContext())
            {
                // Recreate database
                db.Database.EnsureDeleted();
                db.Database.EnsureCreated();

                // Seed database


                db.SaveChanges();
            }

            using (var db = new MyContext())
            {

                var foo = db.Users.OrderBy(e => e.Id).OrderBy(e => e.Id).ToList();

                // Run queries
                var query = db.Users.OrderBy(e => e.Blog.Id2).ThenBy(e => e.Blog.Id2).ToList();
            }

            Console.WriteLine("Program finished.");
        }
    }


    public class MyContext : DbContext
    {
        // Declare DBSets
        public DbSet<User> Users { get; set; }
        public DbSet<Blog> Blogs { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            // Select 1 provider
            optionsBuilder
                .UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=_ModelApp;Trusted_Connection=True;Connect Timeout=5;ConnectRetryCount=0")
                .EnableSensitiveDataLogging()
                .UseLoggerFactory(new LoggerFactory().AddConsole(LogLevel.Trace));
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            // Configure model
            modelBuilder.Entity<Blog>(b =>
            {
                b.HasKey(e => new { e.Id1, e.Id2 });
                b.HasMany(e => e.Posts).WithOne().HasForeignKey(e => new { e.BlogId1, e.BlogId2 });
            });
        }
    }

    public class User
    {
        public int Id { get; set; }
        public Blog Blog { get; set; }
    }

    public class Blog
    {
        public int UserId { get; set; }
        public int Id1 { get; set; }
        public int Id2 { get; set; }
        public List<Post> Posts { get; set; }
    }

    public class Post
    {
        public int Id { get; set; }
        public int BlogId1 { get; set; }
        public int BlogId2 { get; set; }
    }
}
```

query plan

```
 (QueryContext queryContext) => IEnumerable<User> _InterceptExceptions(
          source: IEnumerable<User> _TrackEntities(
              results: IEnumerable<User> _ShapedQuery(
                  queryContext: queryContext, 
                  shaperCommandContext: SelectExpression: 
                      SELECT [e].[Id]
                      FROM [Users] AS [e]
                      LEFT JOIN [Blogs] AS [e.Blog] ON [e].[Id] = [e.Blog].[UserId]
                      ORDER BY [e.Blog].[Id2], [e.Blog].[Id2], 
                  shaper: BufferedOffsetEntityShaper<User>), 
              queryContext: queryContext, 
              entityTrackingInfos: { itemType: User }, 
              entityAccessors: List<Func<User, object>> 
              { 
                  Func<User, User>, 
              }
          ), 
          contextType: EFSampleApp.MyContext, 
          logger: DiagnosticsLogger<Query>, 
          queryContext: queryContext)
```

exception:

```
System.Data.SqlClient.SqlException: A column has been specified more than once in the order by list. Columns in the order by list must be unique.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, SqlDataReader ds)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean asyncWrite, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.BufferlessMoveNext(Boolean buffer)
   at Microsoft.EntityFrameworkCore.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.<_TrackEntities>d__17`2.MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()
   at System.Collections.Generic.List`1.AddEnumerable(IEnumerable`1 enumerable)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
```

**Comment by maumar** [comment]
We should recognize that both order by terms are the same thing, either at QM level or at SelectExpression

**Comment by maumar** [comment]
This is now fixed, possibly as side-effect of b95f23ffa4fbba305b3f0ea4c864fe53f456ad58
